File: backend/app/schemas/watchlist.py
from pydantic import BaseModel
from datetime import datetime
from app.schemas.movie import MovieOut

class WatchListCreate(BaseModel):
    movie_id: int

class WatchListOut(BaseModel):
    user_id: int
    movie_id: int
    added_at: datetime

    class Config:
        from_attributes = True


# Schema necesara pentru a face join-ul cu Movie, astfel incat in frontend sa am acces la Movie Title
class WatchListWithMovieOut(BaseModel):
    user_id: int
    added_at: datetime
    movie: MovieOut 
    
    class Config:
        orm_mode = True

File: android/app/src/main/java/com/app/movieit/data/model/Watchlist.kt
package com.app.movieit.data.model

import com.google.gson.annotations.SerializedName

data class WatchlistCreate(
    @SerializedName("movie_id") val movieId: Int
)

data class WatchlistItem(
    @SerializedName("user_id") val userId: Int,
    @SerializedName("movie_id") val movieId: Int,
    @SerializedName("added_at") val addedAt: String
)

// pentru GET /watchlist/me si /watchlist/user/{id} (WatchListWithMovieOut)
data class WatchlistItemWithMovie(
    @SerializedName("user_id") val userId: Int,
    @SerializedName("added_at") val addedAt: String,
    val movie: Movie
)